config-graphql.js

```diff
@@ -0,0 +1,53 @@
+const namingConventionRule = {
+  InputValueDefinition: { style: 'camelCase' },
+  InputObjectTypeDefinition: { style: 'PascalCase', requiredSuffixes: ['Input'] },
+  FieldDefinition: 'camelCase',
+  'FieldDefinition[parent.name.value=Mutation]': { style: 'camelCase', forbiddenSuffixes: ['Mutation'] },
+  'FieldDefinition[parent.name.value=Query]': { style: 'camelCase', forbiddenSuffixes: ['Query'] },
+  allowLeadingUnderscore: true,
+};
+
+module.exports = {
+  plugins: ['@graphql-eslint'],
+  parser: '@graphql-eslint/eslint-plugin',
+  rules: {
+    '@graphql-eslint/known-type-names': 'warn',
+    '@graphql-eslint/require-deprecation-reason': 'error',
+    '@graphql-eslint/no-scalar-result-type-on-mutation': 'warn',
+    '@graphql-eslint/no-duplicate-fields': 'warn',
+    '@graphql-eslint/naming-convention': ['error', namingConventionRule],
```

Comment:
Since the linted files are auto-generated is it going to be possible to ignore specific violations of the lint rule in services? Ideally people will make the required changes asap but I'm a bit worried if they're going to be forced to do that work immediately after adding this lint config

Reply:
It seems they support that. Let me see how that would work with auto-generated files. 

https://the-guild.dev/graphql/eslint/docs#key-features

Reply:
We can indeed ignore the lint rules if we manage to leave the below type of comments on the graphql file itself.

```
input InputA {
  # eslint-disable-next-line
  field: String;
}
```

The problem is that the code that auto generates the `.graphql` files doesn't parse those types of comments that I add on the `*.schema.ts` files. 

I tried again to read the schema from the `.ts` files directly but had a hard time.

Reply:
I was finally able to get the lint to work directly with the `.ts` files but the problem is that the lint library doesn't work very well with the federation directives/types. I have to pretty much manually write all those definitions in a `.ts` just how they are redefined in every [federated.graphql](https://github.com/neofinancial/savings-account-service/blob/master/lib/admin-federated-schema.graphql#L1) file. 

At this point, I am trying to figure out why the autogenerated .graphql file doesn't include the comments (using `#`) that I leave in the schema. That's the thing that works with the `eslint-disable` approach.